In [11]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

## TASK 1:

In [12]:
cd raw

[WinError 2] The system cannot find the file specified: 'raw'
C:\Users\sofia\Desktop\COSC 301\project-group38-project\data\raw


Pipeline

In [13]:
##loading in data
df = pd.read_csv("weatherstats_kelowna_daily.csv")
df

date  max_temperature  avg_hourly_temperature  avg_temperature  \
0     2021-07-16             32.4                   24.28            23.40   
1     2021-07-15             31.6                   25.70            24.95   
2     2021-07-14             36.3                   26.42            24.95   
3     2021-07-13             35.7                   24.52            23.80   
4     2021-07-12             34.6                   24.88            23.55   
...          ...              ...                     ...              ...   
4995  2007-11-12             10.7                    8.49             8.44   
4996  2007-11-11              8.9                    5.22             4.65   
4997  2007-11-10             12.2                    6.25             8.80   
4998  2007-11-09             10.8                    6.90             6.75   
4999  2007-11-08              6.3                    4.74             3.40   

      min_temperature  max_humidex  min_windchill  max_relative_humidity  \
0                14.4         32.0            NaN                   62.0   
1                18.3         33.0            NaN                   53.0   
2                13.6         35.0            NaN                   69.0   
3                11.9         34.0            NaN                   65.0   
4                12.5         34.0            NaN                   71.0   
...               ...          ...            ...                    ...   
4995              6.2          NaN            NaN                   83.0   
4996              0.4          NaN           -1.0                   84.0   
4997              5.4          NaN           -1.0                   93.0   
4998              2.7          NaN            NaN                   99.0   
4999              0.5          NaN            NaN                   97.0   

      avg_hourly_relative_humidity  avg_relative_humidity  ...  \
0                             35.9                   41.5  ...   
1                             31.7                   38.0  ...   
2                             32.3                   40.5  ...   
3                             37.2                   39.0  ...   
4                             36.5                   43.0  ...   
...                            ...                    ...  ...   
4995                          49.3                   60.0  ...   
4996                          60.0                   62.5  ...   
4997                          67.0                   63.0  ...   
4998                          87.2                   84.5  ...   
4999                          94.0                   91.5  ...   

      avg_cloud_cover_4  min_cloud_cover_4  max_cloud_cover_8  \
0                   2.0                0.0                NaN   
1                   1.5                0.0                NaN   
2                   1.0                0.0                NaN   
3                   0.0                0.0                NaN   
4                   0.0                0.0                NaN   
...                 ...                ...                ...   
4995                NaN                NaN                NaN   
4996                NaN                NaN                NaN   
4997                NaN                NaN                NaN   
4998                NaN                NaN                NaN   
4999                NaN                NaN                NaN   

      avg_hourly_cloud_cover_8  avg_cloud_cover_8  min_cloud_cover_8  \
0                          NaN                NaN                NaN   
1                          NaN                NaN                NaN   
2                          NaN                NaN                NaN   
3                          NaN                NaN                NaN   
4                          NaN                NaN                NaN   
...                        ...                ...                ...   
4995                       NaN                NaN                NaN   
4996        

In [14]:
## Isolating relavant information
df1 = df.drop(columns=['max_humidex', 'min_windchill','max_relative_humidity','avg_hourly_relative_humidity','avg_relative_humidity','avg_cloud_cover_4',
                       'min_cloud_cover_4','max_cloud_cover_8','avg_hourly_cloud_cover_8','avg_cloud_cover_8','min_cloud_cover_8','max_cloud_cover_10',
                       'avg_hourly_cloud_cover_10','max_wind_speed','avg_hourly_wind_speed','avg_wind_speed','min_wind_speed','max_wind_gust','daylight','sunrise_f','sunset_f'
                        ,'avg_cloud_cover_10','avg_cloud_cover_10','min_cloud_cover_10','min_relative_humidity',
                       'max_dew_point','avg_hourly_dew_point','avg_dew_point','min_dew_point','max_cloud_cover_4','avg_hourly_cloud_cover_4',
                      'wind_gust_dir_10s','max_pressure_sea','avg_hourly_pressure_sea','avg_pressure_sea','min_pressure_sea',
                      'max_pressure_station','avg_hourly_pressure_station','avg_pressure_station','min_pressure_station',
                       'max_visibility','avg_hourly_visibility','avg_visibility','min_visibility','max_health_index',
                      'avg_hourly_health_index','avg_health_index','min_health_index','heatdegdays','cooldegdays','growdegdays_5','growdegdays_7','growdegdays_10',
                      'min_uv_forecast','max_uv_forecast','min_high_temperature_forecast','max_high_temperature_forecast','min_low_temperature_forecast','max_low_temperature_forecast','solar_radiation','sunrise','sunset','snow','snow_on_ground','rain'])
df1
##https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.drop.html

date  max_temperature  avg_hourly_temperature  avg_temperature  \
0     2021-07-16             32.4                   24.28            23.40   
1     2021-07-15             31.6                   25.70            24.95   
2     2021-07-14             36.3                   26.42            24.95   
3     2021-07-13             35.7                   24.52            23.80   
4     2021-07-12             34.6                   24.88            23.55   
...          ...              ...                     ...              ...   
4995  2007-11-12             10.7                    8.49             8.44   
4996  2007-11-11              8.9                    5.22             4.65   
4997  2007-11-10             12.2                    6.25             8.80   
4998  2007-11-09             10.8                    6.90             6.75   
4999  2007-11-08              6.3                    4.74             3.40   

      min_temperature  precipitation  
0                14.4            0.0  
1                18.3            0.0  
2                13.6            0.0  
3                11.9            0.0  
4                12.5            0.0  
...               ...            ...  
4995              6.2            1.4  
4996              0.4            0.0  
4997              5.4            2.8  
4998              2.7            0.6  
4999              0.5            3.8  

[5000 rows x 6 columns]

In [15]:
# dropping Nan Values 
df2 = df1.dropna()
df2
##https://www.kite.com/python/answers/how-to-drop-empty-rows-from-a-pandas-dataframe-in-python#:~:text=Use%20df.,contain%20NaN%20under%20those%20columns.

date  max_temperature  avg_hourly_temperature  avg_temperature  \
0     2021-07-16             32.4                   24.28            23.40   
1     2021-07-15             31.6                   25.70            24.95   
2     2021-07-14             36.3                   26.42            24.95   
3     2021-07-13             35.7                   24.52            23.80   
4     2021-07-12             34.6                   24.88            23.55   
...          ...              ...                     ...              ...   
4995  2007-11-12             10.7                    8.49             8.44   
4996  2007-11-11              8.9                    5.22             4.65   
4997  2007-11-10             12.2                    6.25             8.80   
4998  2007-11-09             10.8                    6.90             6.75   
4999  2007-11-08              6.3                    4.74             3.40   

      min_temperature  precipitation  
0                14.4            0.0  
1                18.3            0.0  
2                13.6            0.0  
3                11.9            0.0  
4                12.5            0.0  
...               ...            ...  
4995              6.2            1.4  
4996              0.4            0.0  
4997              5.4            2.8  
4998              2.7            0.6  
4999              0.5            3.8  

[4848 rows x 6 columns]

In [16]:
#resetting index
df3 = df2.reset_index()


In [ ]:
## Seperating into dataframes by year

In [38]:
df2021 = df3[df3.date.str.contains("2021")] 
df2020 = df3[df3.date.str.contains("2020")] 
df2019 = df3[df3.date.str.contains("2019")] 
df2018 = df3[df3.date.str.contains("2018")]
df2017 = df3[df3.date.str.contains("2017")] 
df2016 = df3[df3.date.str.contains("2016")]
df2015 = df3[df3.date.str.contains("2015")] 
df2014 = df3[df3.date.str.contains("2014")]
df2013 = df3[df3.date.str.contains("2013")] 
df2012 = df3[df3.date.str.contains("2012")]
df2011 = df3[df3.date.str.contains("2011")] 
df2010 = df3[df3.date.str.contains("2010")]
df2009 = df3[df3.date.str.contains("2009")] 
df2008 = df3[df3.date.str.contains("2008")]
df2007 = df3[df3.date.str.contains("2007")]

In [ ]:
## Seperating dataframes by monthly data


In [32]:
##2021
df2021jan = df2021[df2021.date.str.contains("2021-01")]
df2021feb = df2021[df2021.date.str.contains("2021-02")]
df2021march = df2021[df2021.date.str.contains("2021-03")]
df2021april = df2021[df2021.date.str.contains("2021-04")]
df2021may = df2021[df2021.date.str.contains("2021-05")]
df2021jun = df2021[df2021.date.str.contains("2021-06")]
df2021jul = df2021[df2021.date.str.contains("2021-07")]

In [35]:
##2020
df2020jan = df2020[df2020.date.str.contains("2020-01")]
df2020feb = df2020[df2020.date.str.contains("2020-02")]
df2020march = df2020[df2020.date.str.contains("2020-03")]
df2020april = df2020[df2020.date.str.contains("2020-04")]
df2020may = df2020[df2020.date.str.contains("2020-05")]
df2020jun = df2020[df2020.date.str.contains("2020-06")]
df2020jul = df2020[df2020.date.str.contains("2020-07")]
df2020aug = df2020[df2020.date.str.contains("2020-08")]
df2020sep = df2020[df2020.date.str.contains("2020-09")]
df2020oct = df2020[df2020.date.str.contains("2020-10")]
df2020nov = df2020[df2020.date.str.contains("2020-11")]
df2020dec = df2020[df2020.date.str.contains("2020-12")]

In [ ]:
#2019
df2019jan = df2019[df2019.date.str.contains("2019-01")]
df2019feb = df2019[df2019.date.str.contains("2019-02")]
df2019march = df2019[df2019.date.str.contains("2019-03")]
df2019april = df2019[df2019.date.str.contains("2019-04")]
df2019may = df2019[df2019.date.str.contains("2019-05")]
df2019jun = df2019[df2019.date.str.contains("2019-06")]
df2019jul = df2019[df2019.date.str.contains("2019-07")]
df2019aug = df2019[df2019.date.str.contains("2019-08")]
df2019sep = df2019[df2019.date.str.contains("2019-09")]
df2019oct = df2019[df2019.date.str.contains("2019-10")]
df2019nov = df2019[df2019.date.str.contains("2019-11")]
df2019dec = df2019[df2019.date.str.contains("2019-12")]

In [ ]:
#2018
df2018jan = df2018[df2018.date.str.contains("2018-01")]
df2018feb = df2018[df2018.date.str.contains("2018-02")]
df2018march = df2018[df2018.date.str.contains("2018-03")]
df2018april = df2018[df2018.date.str.contains("2018-04")]
df2018may = df2018[df2018.date.str.contains("2018-05")]
df2018jun = df2018[df2018.date.str.contains("2018-06")]
df2018jul = df2018[df2018.date.str.contains("2018-07")]
df2018aug = df2018[df2018.date.str.contains("2018-08")]
df2018sep = df2018[df2018.date.str.contains("2018-09")]
df2018oct = df2018[df2018.date.str.contains("2018-10")]
df2018nov = df2018[df2018.date.str.contains("2018-11")]
df2018dec = df2018[df2018.date.str.contains("2018-12")]

In [ ]:
#2017
df2017jan = df2017[df2017.date.str.contains("2017-01")]
df2017feb = df2017[df2017.date.str.contains("2017-02")]
df2017march = df2017[df2017.date.str.contains("2017-03")]
df2017april = df2017[df2017.date.str.contains("2017-04")]
df2017may = df2017[df2017.date.str.contains("2017-05")]
df2017jun = df2017[df2017.date.str.contains("2017-06")]
df2017jul = df2017[df2017.date.str.contains("2017-07")]
df2017aug = df2017[df2017.date.str.contains("2017-08")]
df2017sep = df2017[df2017.date.str.contains("2017-09")]
df2017oct = df2017[df2017.date.str.contains("2017-10")]
df2017nov = df2017[df2017.date.str.contains("2017-11")]
df2017dec = df2017[df2017.date.str.contains("2017-12")]

In [ ]:
#2016
df2016jan = df2016[df2016.date.str.contains("2016-01")]
df2016feb = df2016[df2016.date.str.contains("2016-02")]
df2016march = df2016[df2016.date.str.contains("2016-03")]
df2016april = df2016[df2016.date.str.contains("2016-04")]
df2016may = df2016[df2016.date.str.contains("2016-05")]
df2016jun = df2016[df2016.date.str.contains("2016-06")]
df2016jul = df2016[df2016.date.str.contains("2016-07")]
df2016aug = df2016[df2016.date.str.contains("2016-08")]
df2016sep = df2016[df2016.date.str.contains("2016-09")]
df2016oct = df2016[df2016.date.str.contains("2016-10")]
df2016nov = df2016[df2016.date.str.contains("2016-11")]
df2016dec = df2016[df2016.date.str.contains("2016-12")]

In [36]:
#2015
df2015jan = df2015[df2015.date.str.contains("2015-01")]
df2015feb = df2015[df2015.date.str.contains("2015-02")]
df2015march = df2015[df2015.date.str.contains("2015-03")]
df2015april = df2015[df2015.date.str.contains("2015-04")]
df2015may = df2015[df2015.date.str.contains("2015-05")]
df2015jun = df2015[df2015.date.str.contains("2015-06")]
df2015jul = df2015[df2015.date.str.contains("2015-07")]
df2015aug = df2015[df2015.date.str.contains("2015-08")]
df2015sep = df2015[df2015.date.str.contains("2015-09")]
df2015oct = df2015[df2015.date.str.contains("2015-10")]
df2015nov = df2015[df2015.date.str.contains("2015-11")]
df2015dec = df2015[df2015.date.str.contains("2015-12")]

In [ ]:
#2014
df2014jan = df2014[df2014.date.str.contains("2014-01")]
df2014feb = df2014[df2014.date.str.contains("2014-02")]
df2014march = df2014[df2014.date.str.contains("2014-03")]
df2014april = df2014[df2014.date.str.contains("2014-04")]
df2014may = df2014[df2014.date.str.contains("2014-05")]
df2014jun = df2014[df2014.date.str.contains("2014-06")]
df2014jul = df2014[df2014.date.str.contains("2014-07")]
df2014aug = df2014[df2014.date.str.contains("2014-08")]
df2014sep = df2014[df2014.date.str.contains("2014-09")]
df2014oct = df2014[df2014.date.str.contains("2014-10")]
df2014nov = df2014[df2014.date.str.contains("2014-11")]
df2014dec = df2014[df2014.date.str.contains("2014-12")]

In [ ]:
#2013
df2013jan = df2013[df2013.date.str.contains("2013-01")]
df2013feb = df2013[df2013.date.str.contains("2013-02")]
df2013march = df2013[df2013.date.str.contains("2013-03")]
df2013april = df2013[df2013.date.str.contains("2013-04")]
df2013may = df2013[df2013.date.str.contains("2013-05")]
df2013jun = df2013[df2013.date.str.contains("2013-06")]
df2013jul = df2013[df2013.date.str.contains("2013-07")]
df2013aug = df2013[df2013.date.str.contains("2013-08")]
df2013sep = df2013[df2013.date.str.contains("2013-09")]
df2013oct = df2013[df2013.date.str.contains("2013-10")]
df2013nov = df2013[df2013.date.str.contains("2013-11")]
df2013dec = df2013[df2013.date.str.contains("2013-12")]

In [ ]:
#2012
df2012jan = df2012[df2012.date.str.contains("2012-01")]
df2012feb = df2012[df2012.date.str.contains("2012-02")]
df2012march = df2012[df2012.date.str.contains("2012-03")]
df2012april = df2012[df2012.date.str.contains("2012-04")]
df2012may = df2012[df2012.date.str.contains("2012-05")]
df2012jun = df2012[df2012.date.str.contains("2012-06")]
df2012jul = df2012[df2012.date.str.contains("2012-07")]
df2012aug = df2012[df2012.date.str.contains("2012-08")]
df2012sep = df2012[df2012.date.str.contains("2012-09")]
df2012oct = df2012[df2012.date.str.contains("2012-10")]
df2012nov = df2012[df2012.date.str.contains("2012-11")]
df2012dec = df2012[df2012.date.str.contains("2012-12")]

In [ ]:
#2011
df2011jan = df2011[df2011.date.str.contains("2011-01")]
df2011feb = df2011[df2011.date.str.contains("2011-02")]
df2011march = df2011[df2011.date.str.contains("2011-03")]
df2011april = df2011[df2011.date.str.contains("2011-04")]
df2011may = df2011[df2011.date.str.contains("2011-05")]
df2011jun = df2011[df2011.date.str.contains("2011-06")]
df2011jul = df2011[df2011.date.str.contains("2011-07")]
df2011aug = df2011[df2011.date.str.contains("2011-08")]
df2011sep = df2011[df2011.date.str.contains("2011-09")]
df2011oct = df2011[df2011.date.str.contains("2011-10")]
df2011nov = df2011[df2011.date.str.contains("2011-11")]
df2011dec = df2011[df2011.date.str.contains("2011-12")]

In [ ]:
#2010
df2010jan = df2010[df2010.date.str.contains("2010-01")]
df2010feb = df2010[df2010.date.str.contains("2010-02")]
df2010march = df2010[df2010.date.str.contains("2010-03")]
df2010april = df2010[df2010.date.str.contains("2010-04")]
df2010may = df2010[df2010.date.str.contains("2010-05")]
df2010jun = df2010[df2010.date.str.contains("2010-06")]
df2010jul = df2010[df2010.date.str.contains("2010-07")]
df2010aug = df2010[df2010.date.str.contains("2010-08")]
df2010sep = df2010[df2010.date.str.contains("2010-09")]
df2010oct = df2010[df2010.date.str.contains("2010-10")]
df2010nov = df2010[df2010.date.str.contains("2010-11")]
df2010dec = df2010[df2010.date.str.contains("2010-12")]

In [42]:
#2009
df2009jan = df2009[df2009.date.str.contains("2009-01")]
df2009feb = df2009[df2009.date.str.contains("2009-02")]
df2009march = df2009[df2009.date.str.contains("2009-03")]
df2009april = df2009[df2009.date.str.contains("2009-04")]
df2009may = df2009[df2009.date.str.contains("2009-05")]
df2009jun = df2009[df2009.date.str.contains("2009-06")]
df2009jul = df2009[df2009.date.str.contains("2009-07")]
df2009aug = df2009[df2009.date.str.contains("2009-08")]
df2009sep = df2009[df2009.date.str.contains("2009-09")]
df2009oct = df2009[df2009.date.str.contains("2009-10")]
df2009nov = df2009[df2009.date.str.contains("2009-11")]
df2009dec = df2009[df2009.date.str.contains("2009-12")]

In [45]:
#2008
df2008jan = df2008[df2008.date.str.contains("2008-01")]
df2008feb = df2008[df2008.date.str.contains("2008-02")]
df2008march = df2008[df2008.date.str.contains("2008-03")]
df2008april = df2008[df2008.date.str.contains("2008-04")]
df2008may = df2008[df2008.date.str.contains("2008-05")]
df2008jun = df2008[df2008.date.str.contains("2008-06")]
df2008jul = df2008[df2008.date.str.contains("2008-07")]
df2008aug = df2008[df2008.date.str.contains("2008-08")]
df2008sep = df2008[df2008.date.str.contains("2008-09")]
df2008oct = df2008[df2008.date.str.contains("2008-10")]
df2008nov = df2008[df2008.date.str.contains("2008-11")]
df2008dec = df2008[df2008.date.str.contains("2008-12")]

In [46]:
#2007
df2007nov = df2007[df2007.date.str.contains("2007-11")]
df2007dec = df2007[df2007.date.str.contains("2007-12")]

## TASK 2:

Method chaining

In [ ]:
## Loading data, dropping unwanted columns, handle Nan values, replacing index
df1 = (pd.read_csv("weatherstats_kelowna_daily.csv")
       .drop(columns=['max_humidex', 'min_windchill','max_relative_humidity','avg_hourly_relative_humidity','avg_relative_humidity','avg_cloud_cover_4',
                       'min_cloud_cover_4','max_cloud_cover_8','avg_hourly_cloud_cover_8','avg_cloud_cover_8','min_cloud_cover_8','max_cloud_cover_10',
                       'avg_hourly_cloud_cover_10','max_wind_speed','avg_hourly_wind_speed','avg_wind_speed','min_wind_speed','max_wind_gust','daylight','sunrise_f','sunset_f'
                        ,'avg_cloud_cover_10','avg_cloud_cover_10','min_cloud_cover_10','min_relative_humidity',
                       'max_dew_point','avg_hourly_dew_point','avg_dew_point','min_dew_point','max_cloud_cover_4','avg_hourly_cloud_cover_4',
                      'wind_gust_dir_10s','max_pressure_sea','avg_hourly_pressure_sea','avg_pressure_sea','min_pressure_sea',
                      'max_pressure_station','avg_hourly_pressure_station','avg_pressure_station','min_pressure_station',
                       'max_visibility','avg_hourly_visibility','avg_visibility','min_visibility','max_health_index',
                      'avg_hourly_health_index','avg_health_index','min_health_index','heatdegdays','cooldegdays','growdegdays_5','growdegdays_7','growdegdays_10',
                      'min_uv_forecast','max_uv_forecast','min_high_temperature_forecast','max_high_temperature_forecast','min_low_temperature_forecast','max_low_temperature_forecast'])
        .dropna()
        .reset_index()
      )
df1

In [ ]:
def load_and_process("weatherstats_kelowna_daily.csv"):
   
    ## Loading data, dropping unwanted columns, handle Nan values, replacing index

df1 = (.drop(columns=['max_humidex', 'min_windchill','max_relative_humidity','avg_hourly_relative_humidity','avg_relative_humidity','avg_cloud_cover_4',
                       'min_cloud_cover_4','max_cloud_cover_8','avg_hourly_cloud_cover_8','avg_cloud_cover_8','min_cloud_cover_8','max_cloud_cover_10',
                       'avg_hourly_cloud_cover_10','max_wind_speed','avg_hourly_wind_speed','avg_wind_speed','min_wind_speed','max_wind_gust','daylight','sunrise_f','sunset_f'
                        ,'avg_cloud_cover_10','avg_cloud_cover_10','min_cloud_cover_10','min_relative_humidity',
                       'max_dew_point','avg_hourly_dew_point','avg_dew_point','min_dew_point','max_cloud_cover_4','avg_hourly_cloud_cover_4',
                      'wind_gust_dir_10s','max_pressure_sea','avg_hourly_pressure_sea','avg_pressure_sea','min_pressure_sea',
                      'max_pressure_station','avg_hourly_pressure_station','avg_pressure_station','min_pressure_station',
                       'max_visibility','avg_hourly_visibility','avg_visibility','min_visibility','max_health_index',
                      'avg_hourly_health_index','avg_health_index','min_health_index','heatdegdays','cooldegdays','growdegdays_5','growdegdays_7','growdegdays_10',
                      'min_uv_forecast','max_uv_forecast','min_high_temperature_forecast','max_high_temperature_forecast','min_low_temperature_forecast','max_low_temperature_forecast'])
        .dropna()
        .reset_index()
      )
return df1

## TASK 3:

Numerical / Categorical data at a glance

In [40]:
df3.describe(include = np.number).T

count         mean          std    min      25%  \
index                   4848.0  2536.638614  1447.685052   0.00  1300.75   
max_temperature         4848.0    14.419059    11.129565 -17.70     5.30   
avg_hourly_temperature  4848.0     8.444969     9.138111 -23.48     1.48   
avg_temperature         4848.0     8.325602     9.108972 -24.15     1.44   
min_temperature         4848.0     2.233973     7.693908 -30.60    -2.80   
precipitation           4848.0     0.754682     2.124985   0.00     0.00   

                             50%      75%      max  
index                   2571.500  3785.25  4999.00  
max_temperature           14.000    23.50    44.60  
avg_hourly_temperature     8.160    16.05    33.66  
avg_temperature            8.075    16.00    31.80  
min_temperature            1.900     8.50    22.50  
precipitation              0.000     0.30    30.80

In [47]:
df3.describe(exclude = np.number).T

count unique         top freq
date  4848   4848  2021-07-16    1

In [ ]:
%matplotlib inline
sc = sns.scatterplot(data= df3, x="date", y="max_temperature")
plt.figure(figsize = (15,8))
sc

In [ ]:
%matplotlib inline
bp = sns.barplot(x= df2007["date"], y= df3["max_temperature"], data= df2007, label="2007 Max Temperature")


In [ ]:
%matplotlib inline
bp = sns.barplot(x= df2007["date"], y= df3["precipitation"], data= df2007, label="2007 Precipitation")

In [ ]:
%matplotlib inline
bp = sns.barplot(x= df2007["date"], y= df3["avg_temperature"], data= df2007, label="2007 Average Temperature")

## TASK 4:

Research question: 
How has the climate changed in Kelowna in the past ~14 years?

As seen in the plot on line 151, average temperatures appear to remain relatively consistent throughout most of the recorded years, however there are distinct outliers. The outliers are representative of extreme temperatures. 
Based on the table on line 144, it is shown that the highest temperature recorded in the 2007-2021 date range was 44 degrees Celsius.
Within the dataframe it is clear that this occured this year (2021), during the recent heat wave that occured. 
The table also shows that the lowest recorded temperature in the date range was -30.6 degrees Celsius.
When looking at the dataframe it can be identified that this low occured in December 2008.

